### PR TITLE
"[AN-2460] Option to reset color

### PR DESCRIPTION
**Pull request by ootaisoo**
ColorWrapper added with IntColor and EmptyColor options so far.

### CODE REVIEW

File: lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/runtime/WebMessageChannel.kt [hunk 1]
```diff
@@ -40,6 +41,8 @@ class WebMessageChannel(private val logLevel: LogLevel, ports: List<WebMessagePo
     }
 
     fun sendMessage(bridgeMessage: BridgeMessage) {
+//        serializer = GsonProvider.newInstance()
```
Comment (makedonsky94) [inline lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/runtime/WebMessageChannel.kt:44]: should remove comment here

File: lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/api/serializer/gson/ForceNullAdapterFactory.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import com.google.gson.TypeAdapter
+import com.google.gson.TypeAdapterFactory
+import com.google.gson.reflect.TypeToken
+import com.google.gson.stream.JsonReader
+import com.google.gson.stream.JsonWriter
+import com.tradingview.lightweightcharts.api.series.models.ColorWrapper.IntColor
+import java.io.IOException
+
+
+class ForceNullAdapterFactory: TypeAdapterFactory {
```
Comment (makedonsky94) [inline lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/api/serializer/gson/ForceNullAdapterFactory.kt:14]: Not used anywhere

File: lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/runtime/WebMessageChannel.kt (hunk 1 above)
Comment (ootaisoo) [reply lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/runtime/WebMessageChannel.kt]: removed

File: lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/api/serializer/gson/ForceNullAdapterFactory.kt (hunk 2 above)
Comment (ootaisoo) [reply lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/api/serializer/gson/ForceNullAdapterFactory.kt]: Deleted

File: lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/api/options/models/TimeScaleOptions.kt [hunk 3]
```diff
@@ -13,9 +14,9 @@ data class TimeScaleOptions(
         var lockVisibleTimeRangeOnResize: Boolean? = null,
         var rightBarStaysOnScroll: Boolean? = null,
         var borderVisible: Boolean? = null,
-        @ColorInt
+
         @JsonAdapter(ColorAdapter::class)
```
[the remark this answers is not in the document]
Comment (ootaisoo) [reply lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/api/options/models/TimeScaleOptions.kt]: Registered as regular adapter.

File: lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/runtime/WebMessageChannel.kt [hunk 4]
```diff
@@ -51,7 +52,9 @@ class WebMessageChannel(private val logLevel: LogLevel, ports: List<WebMessagePo
         bridgeMessage: BridgeMessage,
         port: WebMessagePortCompat? = null
     ): WebMessageCompat {
-        return WebMessageCompat(serializer.toJson(bridgeMessage), port?.let { arrayOf(it) })
+        val jsonMessage = serializer.toJson(bridgeMessage)
+        Log.e("jsonMessage", jsonMessage)
```
Comment (makedonsky94) [inline lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/runtime/WebMessageChannel.kt:56]: remove the log

File: lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/api/series/models/IntColor.kt [hunk 5]
```diff
@@ -5,32 +5,49 @@ import com.google.gson.*
...
     }
 }
 
-private fun IntColor.toHexString(): String {
+fun Int.toIntColor(): IntColor {
+    return IntColor(this)
+}
+
+fun String.toIntColor(): IntColor {
+    return IntColor(this.toColor()!!)
```
Comment (makedonsky94) [inline lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/api/series/models/IntColor.kt:47]: do we need !! operator?

Comment (makedonsky94) [reply lightweightlibrary/src/main/java/com/tradingview/lightweightcharts/api/series/models/IntColor.kt:47]: Also throws JsonParseException which is not acceptable for this method

**Pull request merged**